Bluetooth: Mesh: Fixes seg_tx_reset adv buf unref

Campou opened the pull request:
In seg_tx_reset() in transport.c, set the busy flag to 0U
before doing adv buf unref, which will avoid sending
unnecessary adv packets in case the adv buf is already put
in the mesh adv_queue.

Fixes: #20970

Signed-off-by: llymaximus <llymaximus@gmail.com>

jhedberg commented:
@Campou you have to submit the PR against master. It seems your local tree is still stuck in 1.14.x. The entire mesh code moved from `subsys/bluetooth/host/mesh` to `subsys/bluetooth/mesh` since then. Probably the patch is easy to forward-port with only the path changing.

Campou commented:
@jhedberg Thanks.

jhedberg commented:
@Campou looks good, but could you please fix the gitlint error regarding author information (and signed-off-by line) format? Merging will be prevented until it's fixed.

Commit:
Bluetooth: Mesh: Fixes seg_tx_reset adv buf unref

In seg_tx_reset() in transport.c, set the busy flag to 0U
before doing adv buf unref, which will avoid sending
unnecessary adv packets in case the adv buf is already put
in the mesh adv_queue.

Fixes #20970

Signed-off-by: Maximus Liu <maximus.liu@gmail.com>

Campou commented:
@jhedberg Done, and thanks.
